experimental/hip/dynamic_symbols.c

```diff
@@ -49,24 +50,112 @@ static iree_status_t iree_hal_hip_dynamic_symbols_resolve_all(
   return iree_ok_status();
 }
 
+static bool iree_hal_hip_try_load_dylib(const char* file_path,
+                                        iree_dynamic_library_flags_t flags,
+                                        iree_allocator_t allocator,
+                                        iree_status_t* pending_fail_status,
+                                        iree_dynamic_library_t** out_library) {
+  iree_status_t status = iree_dynamic_library_load_from_file(
+      file_path, flags, allocator, out_library);
+  if (iree_status_is_ok(status)) {
+    return true;
+  }
+
+  char* buffer = NULL;
+  iree_host_size_t length = 0;
+  if (iree_status_to_string(status, &allocator, &buffer, &length)) {
+    *pending_fail_status = iree_status_annotate_f(
+        *pending_fail_status, "\n  Tried: %s\n    %s", file_path, buffer);
```

Comment:
status messages aren't guaranteed to have newlines and such preserved everywhere, or different annotations can be rendered differently (with newlines between them) - only include newlines within a single annotation/message

Reply:
Switched to string builder.